### PR TITLE
Create Order service

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: src/main/java/mate/academy/Main.java [hunk 1]
```diff
@@ -44,12 +53,29 @@ public static void main(String[] args) {
...
         System.out.println(movieSessionService.get(yesterdayMovieSession.getId()));
-        System.out.println(movieSessionService.findAvailableSessions(
-                        fastAndFurious.getId(), LocalDate.now()));
+
+        User user = new User();
+        user.setEmail("nikita@gmail.com");
+        user.setSalt(HashUtil.getSalt());
+        user.setPassword(HashUtil.hashPassword("1234Qwerty", user.getSalt()));
+        UserService userService = (UserService) injector.getInstance(UserService.class);
+        userService.add(user);
```
Comment (Dimagaa) [inline src/main/java/mate/academy/Main.java:68]: You have AuthentificationService to register the user
Within the UserService, the user password is hashed during the execution of the add() method

File: src/main/java/mate/academy/Main.java [hunk 2]
```diff
@@ -44,12 +53,29 @@ public static void main(String[] args) {
...
+
+        ShoppingCartService shoppingCartService = (ShoppingCartService) injector
+                .getInstance(ShoppingCartService.class);
+        shoppingCartService.registerNewShoppingCart(user);
```
Comment (Dimagaa) [inline src/main/java/mate/academy/Main.java:72]: AuthentificationService  handles it

**Review by Dimagaa — APPROVED** [review]
Well done!

File: src/main/java/mate/academy/Main.java (hunk 1 above)
Comment (nacenik) [reply src/main/java/mate/academy/Main.java:68]: good point, fix it

File: src/main/java/mate/academy/Main.java (hunk 2 above)
Comment (nacenik) [reply src/main/java/mate/academy/Main.java:72]: Good point

File: src/main/java/mate/academy/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToMany
+    @JoinColumn(name = "order_id")
+    private List<Ticket> tickets;
+    @Column(name = "order_time")
+    private LocalDateTime orderDate;
```
Comment (nacenik) [inline src/main/java/mate/academy/model/Order.java:27]: order date or order time?

File: src/main/java/mate/academy/dao/impl/OrderDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert a order : " + order, e);
```
Comment (nacenik) [inline src/main/java/mate/academy/dao/impl/OrderDaoImpl.java:30]: Override `toString` method for Order class.

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution, let's improve it :)

File: src/main/java/mate/academy/model/Order.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToMany
+    @JoinColumn(name = "order_id")
+    private List<Ticket> tickets;
+    @Column(name = "order_time")
+    private LocalDateTime orderTime;
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/model/Order.java:27]: It is actually dateTime, not just date and not just time

File: src/main/java/mate/academy/service/impl/OrderSeviceImpl.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        List<Ticket> tickets = new ArrayList<>(shoppingCart.getTickets());
+        Order newOrder = new Order();
+        newOrder.setUser(shoppingCart.getUser());
+        newOrder.setTickets(tickets);
+        newOrder.setOrderTime(LocalDateTime.now());
+        shoppingCart.getTickets().clear();
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/service/impl/OrderSeviceImpl.java:30]: It is better to move shopping cart cleanup logic into separate method

Comment (andrii-hoienko) [inline src/main/java/mate/academy/service/impl/OrderSeviceImpl.java:30]: code duplication (shoppingCart.getTickets())

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, but some minor improvements required

File: src/main/java/mate/academy/service/impl/OrderSeviceImpl.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order newOrder = createOrder(shoppingCart);
+        shoppingCartDao.update(shoppingCart);
+        return orderDao.add(newOrder);
+    }
+
+    private Order createOrder(ShoppingCart shoppingCart) {
```
Comment (nacenik) [inline src/main/java/mate/academy/service/impl/OrderSeviceImpl.java:30]: private methods should be after publics

**Review by nacenik — APPROVED** [review]
LGTM! 👍

**Pull request closed without merging**